nornir/plugins/inventory/ansible.py

```diff
@@ -242,24 +253,178 @@ def load_hosts_file(self) -> None:
             self.original_data = cast(AnsibleGroupsDict, YAML.load(f))
 
 
-def parse(hostsfile: str) -> Tuple[HostsDict, GroupsDict, DefaultsDict]:
-    try:
-        parser: AnsibleParser = INIParser(hostsfile)
-    except cp.Error:
+class ScriptParser(AnsibleParser):
+    def verify_file(self) -> bool:
+        with open(self.hostsfile, "rb") as inv_file:
+            initial_chars = inv_file.read(2)
+            if initial_chars.startswith(b"#!") and os.access(self.hostsfile, os.X_OK):
+                return True
+        return False
+
+    def load_hosts_file(self) -> None:
+        if not self.verify_file():
+            raise TypeError(f"AnsibleInventory: invalid script file {self.hostsfile}")
+
+        proc = subprocess.Popen(
+            [self.hostsfile, "--list"], stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        std_out, std_err = proc.communicate()
+
+        if proc.returncode != 0:
+            raise OSError(
+                f"AnsibleInventory: {self.hostsfile} exited with non-zero return code"
+            )
+
+        processed = json.loads(std_out.decode())
+
+        self.original_data = self.normalize(processed)
+
+    def normalize(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        groups: DefaultDict[str, Dict[str, Any]] = defaultdict(dict)
+        # Dict[str, AnsibleGroupDataDict] does not work because of
+        # https://github.com/python/mypy/issues/5359
+        result: Dict[str, Dict[str, Dict[str, Dict[str, Any]]]] = {
+            "all": {"children": groups}
+        }
+
+        # hostvars are stored in ["_meta"]["hostvars"] if present
+        hostvars = data.get("_meta", {}).get("hostvars", None)
+
+        if "all" in data.keys():
+            data = data["all"]
+        if "vars" in data.keys():
+            groups["defaults"]["vars"] = data.pop("vars")
+
+        for group, gdata in data.items():
+            if "vars" in gdata.keys():
+                groups[group]["vars"] = gdata["vars"]
+            if "children" in gdata.keys():
+                groups[group]["children"] = {}
+                for child in gdata["children"]:
+                    groups[group]["children"][child] = {}
+            if "hosts" in gdata.keys():
+                groups[group]["hosts"] = {}
+                for host in gdata["hosts"]:
+                    groups[group]["hosts"][host] = hostvars.get(host, None)
+        return result
+
+
+def parse(
+    hostsfile: str, inventory: str = "", hash_behavior: str = "replace"
+) -> Tuple[HostsDict, GroupsDict, DefaultsDict]:
+    possible_sources: List[str] = []
+
+    # add hostsfile to to sources to preserve existing functionality
+    if Path(hostsfile).expanduser().is_file():
+        possible_sources.append(hostsfile)
+
+    inventory_locations = inventory.split(",") if inventory else []
```

Comment:
I am confused by this, why are you splitting this using the `,` as separator? What is the use case and reasoning? This wasn't supported before.

Reply:
In Ansible you are able to pass a list of inventories -- a very very handy feature. This is intended to mimic this. I did *not* do this on `hostsfile` to ensure I didn't break any current functionality/confuse people (that's basically the main reason for the second `inventory` argument). I think it'd be nice to just pass a list, but I assume that wouldn't work if this option was passed as an environment variable so I just rolled with comma sep string instead.